tests/bluetooth/tester/src/gap.c

```diff
@@ -171,11 +172,23 @@ static void le_security_changed(struct bt_conn *conn, bt_security_t level,
 			    CONTROLLER_INDEX, (uint8_t *) &sec_ev, sizeof(sec_ev));
 		break;
 	case BT_SECURITY_ERR_PIN_OR_KEY_MISSING:
-		memcpy(bond_ev.address, addr->a.val, sizeof(bond_ev.address));
-		bond_ev.address_type = addr->type;
+		/* for central role this means that peer have no LTK when we
+		 * started encryption procedure
+		 *
+		 * This means bond is lost and we restart pairing to re-bond
+		 */
+		if (bt_conn_get_info(conn, &info) == 0 &&
+		    info.role == BT_CONN_ROLE_CENTRAL) {
+			LOG_DBG("Bond lost");
+
+			(void)memcpy(bond_ev.address, addr->a.val, sizeof(bond_ev.address));
+			bond_ev.address_type = addr->type;
 
-		tester_send(BTP_SERVICE_ID_GAP, GAP_EV_BOND_LOST,
-			    CONTROLLER_INDEX, (uint8_t *) &bond_ev, sizeof(bond_ev));
+			tester_send(BTP_SERVICE_ID_GAP, GAP_EV_BOND_LOST,
+				    CONTROLLER_INDEX, (uint8_t *)&bond_ev, sizeof(bond_ev));
+
+			bt_conn_set_security(conn, BT_SECURITY_L2 | BT_SECURITY_FORCE_PAIR);
```

Comment:
```suggestion
			(void)bt_conn_set_security(conn, BT_SECURITY_L2 | BT_SECURITY_FORCE_PAIR);
```
or store the result and log error if any.